modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Engine.scala

```diff
@@ -0,0 +1,27 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * A Map of `Sequence`s.
+  */
+case class Engine[+A](sequences: Map[Sequence.Id, Sequence[A]])
```

Comment:
> Why's that you need variance here?
Theoretically I still struggling to understand subtyping in Scala, so I don't really know why it's required, but in practice I needed it for the Functor instance. We don't use the Functor instance anywhere yet, but I carried it over from the previous `Queue` data type. It may come handy if we want to map over all existing `Execution`s or to create a `Foldable` instance.

> Is the idea that we'd add more components to engine, like Observing conditions?
Yes, I'll do it right away in another PR. When I started I had the operator in mind but it turns out the conditions may be an easier first global parameter.